src/dvc_data/diff.py

```diff
@@ -70,28 +70,30 @@ def stats(self) -> Dict[str, int]:
 ROOT = ("",)
 
 
-def diff(
+def diff(  # noqa: C901
```

Comment:
`diff` is getting complex, we can start by extracting inner functions. For now, I am just slapping a _ignore_, and moving on.